### PR TITLE
Analytic Einasto and Hernquist normalization

### PR DESCRIPTION
**Pull request by hsinfan1996**
Some functions in `src/ccl_haloprofile.c` and `pyccl/ccl_haloprofile.i` are not needed anymore, but I did not touch them.

### CODE REVIEW

**Comment by damonge** [comment]
Thanks a lot @hsinfan1996 . Can you check how much faster (or slower) it is to call the gamma functions than computing the normalisation numerically?

**Comment by hsinfan1996** [comment]
> Thanks a lot @hsinfan1996 . Can you check how much faster (or slower) it is to call the gamma functions than computing the normalisation numerically?

1. If `M`, `R_s`, `R_M` and `alpha` are floats, ~9 times faster.
2. If `M`, `R_s`, `R_M` and `alpha` are arrays with each having only one element, ~1.2 times faster.
3. If `M`, `R_s`, `R_M` and `alpha` are arrays with each having 100 elements, ~7 times faster.

**Comment by damonge** [comment]
OK, awesome, in that case feel free to remove all traces of the old functions in `src/ccl_haloprofile.c` and `pyccl/ccl_haloprofile.i`

**Comment by hsinfan1996** [comment]
Coveralls is down.

**Comment by damonge** [comment]
Ignore coveralls. Coverage went down because you reduced the number of lines in the code, I bet.

File: include/ccl.h
Comment (damonge) [inline include/ccl.h:44]: Should you also delete the `ccl_haloprofile.h` file?

**Review by damonge — COMMENTED** [review]
Thanks a lot! It feels great to get rid of unnecessary C code!
Just one comment.

**Comment by damonge** [comment]
It seems coveralls is undergoing some maintenance. I'm sure it's irrelevant, but let's wait for tomorrow before merging (pressing the red merge button makes my brain scream a bit)

**Comment by hsinfan1996** [comment]
> It seems coveralls is undergoing some maintenance. I'm sure it's irrelevant, but let's wait for tomorrow before merging (pressing the red merge button makes my brain scream a bit)

Sure. I want it to pass all the checks too. I will re-run the jobs when Coveralls is stable.

**Review by damonge — APPROVED** [review]
LGTM!

**Pull request merged**
